From e25dbfe95302eeaa1a03a828d05c09479574488a Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Mon, 11 Jun 2018 18:17:54 +0800
Subject: [PATCH] MIPS: Use '+=" instead of '=' to avoid the CFLAGS override

We used the CFLAGS_xxx to workaround the gcc 8 build warnings
for some specific file. But CFLAGS_xxx is also used with '=' in
other places of this Makefile. This override the gcc 8 workaround,
so replace all the '=' with '+=" to fix this issue.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 arch/mips/kernel/Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/mips/kernel/Makefile b/arch/mips/kernel/Makefile
index 795aa4c9ef7f..c97388229391 100644
--- a/arch/mips/kernel/Makefile
+++ b/arch/mips/kernel/Makefile
@@ -128,11 +128,11 @@ obj-$(CONFIG_MIPS_CPS_PM)	+= pm-cps.o
 ifeq ($(CONFIG_CPU_MIPSR2), y)
 CFLAGS_DSP 			= -DHAVE_AS_DSP
 
-CFLAGS_signal.o			= $(CFLAGS_DSP)
-CFLAGS_signal32.o		= $(CFLAGS_DSP)
-CFLAGS_process.o		= $(CFLAGS_DSP)
-CFLAGS_branch.o			= $(CFLAGS_DSP)
-CFLAGS_ptrace.o			= $(CFLAGS_DSP)
+CFLAGS_signal.o			+= $(CFLAGS_DSP)
+CFLAGS_signal32.o		+= $(CFLAGS_DSP)
+CFLAGS_process.o		+= $(CFLAGS_DSP)
+CFLAGS_branch.o			+= $(CFLAGS_DSP)
+CFLAGS_ptrace.o			+= $(CFLAGS_DSP)
 endif
 
 CPPFLAGS_vmlinux.lds		:= $(KBUILD_CFLAGS)
-- 
2.5.0

